### PR TITLE
Fix strip on non-x86 targets

### LINKED ISSUE #8: Cannot strip ARM binaries
I am using this action to cross-compile a CLI application (called `hoard`) for multiple platforms. When compiling for the `armv7-linux-androideabi` target, I got the following error:

```
strip: Unable to recognise the format of the input file `hoard'
```

I am not sure if there is a way to strip these binaries or if stripping should be skipped for them, but either solution would be nice so I can distribute downloads for Android platforms.

### PR DESCRIPTION
**Pull request by taiki-e**
There are dedicated stripping tools for architectures other than x86. If they are installed, use them; otherwise, skip strip.

Fixes #8

cc @Shadow53

### CODE REVIEW

**Comment by Shadow53** [comment]
This looks like it would do the job! Thanks for being so quick about this.

A question, out of curiosity: why run `*-strip --version` instead of something like `which *-strip` or I think `command -q strip` (running the program vs just searching for it)?

**Comment by Shadow53** [comment]
I am trying this branch in my CI to test, and I am still getting the same error:

```
 strip: Unable to recognise the format of the input file `hoard'
```

This is with the following Rust toolchains:

- aarch64-unknown-linux-gnu
- aarch64-unknown-linux-musl
- arm-unknown-linux-gnueabihf
- arm-unknown-linux-musleabihf

Notably, the following successfully built:

- aarch64-linux-android
- arm-linux-androideabi
- armv7-linux-androideabi

It looks like the successful ones were not stripped, as `aarch64-linux-android` does not match the pattern `aarch64*-unknown-linux-*` and I did not have the right cross-compilation tools installed for the other two.

I will update with another comment if I figure anything else out.

File: main.sh [hunk 1]
```diff
@@ -65,6 +63,36 @@ case "${OSTYPE}" in
...
+        if arm-linux-gnueabihf-strip --version &>/dev/null; then
+            strip="arm-linux-gnueabihf-strip"
+        fi
+        ;;
+    arm*-none-eabi | thumb*-none-eabi)
+        if arm-none-eabi-strip --version &>/dev/null; then
+            strip="arm-none-eabi-strip"
+        fi
+        ;;
+    aarch64*-unknown-linux-*)
```
Comment (Shadow53) [inline main.sh:88]: The Android aarch64 target does not contain `-unknown`

**Review by Shadow53 — CHANGES_REQUESTED** [review]

**Comment by Shadow53** [comment]
Okay, I think I found the issue, You set `strip="..."` in the `case` statement, but around line 111 you call `strip` instead of `$strip`. Change it to the latter, and I think the issue I saw will go away.

**Comment by taiki-e** [comment]
Thanks @Shadow53! I've applied both fixes.

**Comment by taiki-e** [comment]
> A question, out of curiosity: why run `*-strip --version` instead of something like `which *-strip` or I think `command -q strip` (running the program vs just searching for it)?

I tried to choose the one that worked most reliably because their subtle differences in behavior are complicated (https://github.com/koalaman/shellcheck/issues/1162), but in fact `strip --version` did not work on macos, so I switched to `which` in latest commit.

**Comment by taiki-e** [comment]
I've tested that these work (https://github.com/taiki-e/test/actions/runs/1286975387). merging

**Pull request merged**